packages/workflow/src/workflow.ts

```diff
@@ -1168,20 +1157,48 @@ export function setHandler<Ret, Args extends any[], T extends SignalDefinition<A
   const activator = getActivator();
   if (def.type === 'signal') {
     activator.signalHandlers.set(def.name, handler as any);
-    const bufferedSignals = activator.bufferedSignals.get(def.name);
-    if (bufferedSignals !== undefined && handler !== undefined) {
-      activator.bufferedSignals.delete(def.name);
-      for (const signal of bufferedSignals) {
-        activator.signalWorkflow(signal);
-      }
-    }
+    if (handler !== undefined) dispatchBufferedSignals();
   } else if (def.type === 'query') {
     activator.queryHandlers.set(def.name, handler as any);
   } else {
     throw new TypeError(`Invalid definition type: ${(def as any).type}`);
   }
 }
 
+/**
+ * Set a signal handler function that will handle signals calls for non-registered signal names.
+ *
+ * Signals are dispatched to the default signal handler in the order that they were accepted by the server.
+ *
+ * If this function is called multiple times for a given signal or query name the last handler will overwrite any previous calls.
+ *
+ * @param handler a function that will handle signals for non-registered signal names, or `undefined` to unset the handler.
+ */
+export function setDefaultSignalHandler(handler: DefaultSignalHandler | undefined): void {
+  const activator = getActivator();
+  activator.defaultSignalHandler = handler;
+  if (handler !== undefined) dispatchBufferedSignals();
```

Comment:
JS people love their `null`s.

```suggestion
  if (handler != null) dispatchBufferedSignals();
```